docs/source/auth/index.rst

```diff
@@ -0,0 +1,639 @@
+
+Admin Users
+-----------
+
+Admin users have unrestricted access to all MLflow resources,
+**including creating or deleting users, updating password and admin status of other users,
+grating or revoking permissions from other users, and managing permissions for all 
+MLflow resources,** even if ``NO_PERMISSIONS`` is explicitly set to an admin.
+
+MLflow has a built-in admin user that will be created the first time MLflow authentication feature is enabled.
+The default admin user credentials are as follows.
+It is recommended that you update the password as soon as possible after it is created.
+
+.. list-table::
+   :widths: 10 10
+   :header-rows: 1
+
+   * - Username
+     - Password
+   * - ``admin``
+     - ``password``
+
+
+Multiple admin users can exist by promoting other users to admin, using the ``2.0/mlflow/users/update-admin`` endpoint.
+
+.. code-block:: python
+    :caption: Example
+
+    import os
+    from mlflow.server import get_app_client
+
+    # authenticate as built-in admin user
+    os.environ["MLFLOW_TRACKING_USERNAME"] = "admin"
+    os.environ["MLFLOW_TRACKING_PASSWORD"] = "password"
+    tracking_uri = "http://localhost:5000/"
+
+    auth_client = get_app_client("basic-auth", tracking_uri=tracking_uri)
+    auth_client.create_user(username="user1", password="pw1")
+    auth_client.update_user_admin(username="user1", is_admin=True)
+
+
+Managing Permissions
+--------------------
+
+MLflow provides :ref:`REST APIs <mlflowAuthServiceCreateUser>` and a client class 
+:py:func:`AuthServiceClient<mlflow.server.auth.client.AuthServiceClient>` to manage users and permissions.
+To instantiate ``AuthServiceClient``, it is recommended that you use :py:func:`mlflow.server.get_app_client`.
+
+.. code-block:: python
+    :caption: Example
+
+    import os
+    from mlflow import MlflowClient
+    from mlflow.server import get_app_client
+
+    tracking_uri = "http://localhost:5000/"
+
+    auth_client = get_app_client("basic-auth", tracking_uri=tracking_uri)
+    auth_client.create_user(username="user1", password="pw1")
+    auth_client.create_user(username="user2", password="pw2")
+
+    os.environ["MLFLOW_TRACKING_USERNAME"] = "user1"
+    os.environ["MLFLOW_TRACKING_PASSWORD"] = "pw1"
+
+    client = MlflowClient(tracking_uri=tracking_uri)
+    experiment_id = client.create_experiment(name="experiment")
+
+    auth_client.create_experiment_permission(
+        experiment_id=experiment_id, username="user2", permission="MANAGE"
+    )
+
+
+Authenticating to MLflow
+========================
+
+Using MLflow UI
+---------------
+
+When a user first visits the MLflow UI on a browser, they will be prompted to login. 
+There is no limit to how many login attempts can be made.
+
+Currently, MLflow UI does not display any information of the current user.
+Once a user is logged in, the only way to log out is to close the browser.
+
+    .. image:: ../_static/images/auth_prompt.png
+
+Using REST API
```

Comment:
Can we modify the order of sub-sections in this section?

- Environment variables
- Credentials File
- REST API

Reply:
okay